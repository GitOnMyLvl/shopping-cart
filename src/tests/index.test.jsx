import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter,  Routes, Route } from "react-router-dom";
import Index from "../routes/index";
import Store from "../routes/store";
import userEvent from "@testing-library/user-event";

describe("Index component", () => {
  it("renders page", () => {
    const { container } = render(
      <MemoryRouter>
        <Index />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders shop after click", async () => {
    const user = userEvent.setup();
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<Index />} />
          <Route path="/store" element={<Store />} />
        </Routes>
      </MemoryRouter>
    );
    const link = screen.getByRole("link", { name: /newest offers/i });
    await user.click(link);
    expect(screen.getByText(/loading/i)).toBeInTheDocument();
  })
});
