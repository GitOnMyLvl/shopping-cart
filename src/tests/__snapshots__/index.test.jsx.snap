// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index component > renders page 1`] = `
<div>
  <section
    class="max-container hero"
  >
    <h1>
      Welcome to the 
      <span
        class="orange-text"
      >
        Super Store
      </span>
    </h1>
    <p>
      Check ou our 
      <a
        class="orange-text"
        href="/store"
      >
        newest offers
      </a>
    </p>
  </section>
</div>
`;
